backend/gateway/gateway.go

```diff
@@ -88,6 +89,14 @@ func RunWithConfig(f *Flags, cfg *gatewayv1.Config, cf *ComponentFactory, assets
 	initScope := scope.SubScope("gateway")
 	initScope.Counter("start").Inc(1)
 
+	// Create the error interceptor so services can register error interceptors if desired.
+	var unaryErrorMiddleware *errorintercept.Middleware
+	if m, err := errorintercept.New(); err != nil {
+		logger.Fatal("could not create error interceptor middleware", zap.Error(err))
+	} else {
+		unaryErrorMiddleware = m.(*errorintercept.Middleware)
+	}
```

Comment:
nit: fatal logs should call `os.Exit(1)` we probably don't need the else unless we change log level.